### ISSUE TITLE
The future of styling widgets (new package)

### ISSUE THREAD

**Issue by ReinBentdal** [ReinBentdal/division#47]
I have made a new package called `styled_widget` which uses the new dart extension feature added in dart version 2.6.0. This is just an early concept.

Link to package: https://github.com/ReinBentdal/styled_widget

What do you think? Is there a future in styling widgets this way? If you like the concept or dont, let me know! Of course contributions are very welcome.

**Comment by moraispgsi** [comment]
The problem I see with this approach is that you cannot hoist/abstract the styling like you would in CSS. Maybe exploring the decorator pattern for it, who knows.

**Comment by ReinBentdal** [comment]
Yes its not possible to create style like in `Division`. I assume the flutter team wants us to use `ThemeData` to style general widgets. This only works for some general widgets like `Card`, `AppBar`, `Button`, etc, but maybe that is enough. 

This could be a way to decouple style
```dart
final Styled textStyle = Styled.text() // StyledText class
  .backgroundColor(Colors.blue)

final Styled iconStyle = Styled.icon() // StyledIcon class
  .iconSize(24)
  .alignment(Alignment.center)

final Styled widgetStyle = Styled.widget() // Styled [widget] class
  .bordeRadius(all: 10)

Widget build(BuildContext context) {
...
Text('some text').style(textStyle),
Icon(Icons.icon).style(iconStyle),
Container().style(widgetStyle)
...
}
```

**Comment by moraispgsi** [comment]
@ReinBentdal This looks like the way it is supposed to be. It is inviting for any React.js enthusiast.

**Comment by moraispgsi** [comment]
@ReinBentdal I would like to help you out in this direction, it looks promising. I don't have much time on my hands but I can help. Just lead the way.
